crates/networking/p2p/rlpx/connection.rs

```diff
@@ -366,7 +326,24 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
                                 negotiated_snap_version = cap.version;
                             }
                         }
-                        _ => {}
+                        "based" => {
+                            let l2_state = L2ConnectedState {
+                                latest_block_sent: 0,
+                                latest_block_added: 0,
+                                blocks_on_queue: BTreeMap::new(),
+                                latest_batch_sent: 0,
+                                store_rollup: StoreRollup::default(),
+                                committer_key: None,
+                                next_block_broadcast: Instant::now()
+                                    + PERIODIC_BLOCK_BROADCAST_INTERVAL,
+                                next_batch_broadcast: Instant::now()
+                                    + PERIODIC_BATCH_BROADCAST_INTERVAL,
+                            };
+                            self.l2_state = L2ConnState::Connected(l2_state);
+                        }
+                        unknown_protocol => {
+                            warn!("Peer sent an unsupported protocol: {unknown_protocol}")
+                        }
```

Comment:
Missing the p2p capability, you could add something like this:
```suggestion
                        "p2p" => {}
                        unknown_protocol => {
                            warn!("Peer sent an unsupported protocol: {unknown_protocol}")
                        }
```

Reply:
Fixed!